lib/orangeade/generator/boolean.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Orangeade.Generator.Boolean do
+  @moduledoc """
+  A boolean values generator {:true, :false, :nil}
+  
+  
+  ## Examples
+
+      iex(1)> Caffeine.Stream.take(
+      ...(1)> Orangeade.Generator.Boolean.stream(),
+      ...(1)> 5_000)
+      [false, false, nil, nil, nil, nil, false, true, false, nil, nil, nil, nil, true,
+      true, nil, nil, nil, nil, false, nil, false, nil, false, nil, false, true,
+      false, true, false, true, false, nil, false, true, false, false, false, false,
+      false, false, true, nil, false, false, false, false, nil, true, true, ...]
+
+  """
+  def stream do
+    alias Orangeade.Generator.BoundNatural, as: BoundNatural
+    BoundNatural.stream(limit: 3)
+      |> Caffeine.Stream.map(&bool/1)
+    end
+    
+    defp bool(e) do
+      cond do
+        e === 0 ->
+          :true
+        e === 1 ->
+          :false
+        e === 2 ->
+          :nil
```

Comment:
This is a great idea for a generator. I never considered `nil` alongside `true` and `false`. We need a different name for the generator though. `Boolean` doesn't fully capture the idea of a truthy or falsey value... the only other name I've come up with is `Logical` which isn't much of an improvement. Perhaps you have a better naming idea